src/hotspot/cpu/aarch64/continuationHelper_aarch64.inline.hpp

```diff
@@ -68,6 +68,14 @@ inline void ContinuationHelper::push_pd(const frame& f) {
   *(intptr_t**)(f.sp() - frame::sender_sp_offset) = f.fp();
 }
 
+inline address ContinuationHelper::return_pc_at(intptr_t* sp) {
+  return pauth_strip_pointer(*(address*)sp);
+}
+
+inline void ContinuationHelper::patch_pc_at(intptr_t* sp, address pc) {
+  *(address*)sp = pauth_sign_return_address(pc);
```

Comment:
This is a bad name. We're not patching the PC.

Reply:
Thanks for your code review.

From `ContinuationHelper::Frame::patch_pc(const frame& f, address pc)` and `void frame::patch_pc(Thread* thread, address pc)`, I thought "patch_pc" means replacing the original PC with the new one, i.e. the argument PC.

In this function `ContinuationHelper::patch_pc_at(intptr_t* sp, address pc)`, I followed the notion that **updating the original PC at address SP with the argument PC**. That's why I used "patch_pc_at".

If you think it's not a good name, how about `set_pc_at()`?
Thanks.

Reply:
> Thanks for your code review.
> 
> From `ContinuationHelper::Frame::patch_pc(const frame& f, address pc)` and `void frame::patch_pc(Thread* thread, address pc)`, I thought "patch_pc" means replacing the original PC with the new one, i.e. the argument PC.
> 
> In this function `ContinuationHelper::patch_pc_at(intptr_t* sp, address pc)`, I followed the notion that **updating the original PC at address SP with the argument PC**. That's why I used "patch_pc_at".
> 
> If you think it's not a good name, how about `set_pc_at()`? Thanks.

A Program Counter is a physical thing. It's made of silicon and metal. It contains an address.

`patch_return_address_at()` would be fine.

Reply:
Updated in the latest revision. Thanks.